### PR TITLE
[wsu] Run kube-proxy as a Windows service

### PR DESCRIPTION
**Pull request by sebsoto**
This commit runs kube-proxy as Windows service, allowing for external
sources to network with pods.

### CODE REVIEW

File: internal/test/wsu/wsu_test.go [hunk 1]
```diff
@@ -322,6 +308,35 @@ func testEastWestNetworking(t *testing.T) {
 	// TODO: test Windows <-> Windows on different node
 }
 
+// deployWindowsServer creates a deployment with a single Windows Server pod, listening on port 80
+func deployWindowsServer() (*appsv1.Deployment, error) {
```
Comment (aravindhp) [inline internal/test/wsu/wsu_test.go]: Please rename to `deployWindowsWebServer()` to indicate the type of Windows application. `WindowsServer` makes me think it is just a Windows Server.

File: tools/ansible/tasks/wsu/powershell/hns.psm1 [hunk 2]
```diff
@@ -0,0 +1,511 @@
+# Code from https://github.com/microsoft/SDN/
```
Comment (aravindhp) [inline tools/ansible/tasks/wsu/powershell/hns.psm1]: Please link to the file.

Does this pick just what we need or did you copy the whole file?

File: tools/ansible/tasks/wsu/powershell/hns.psm1 [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+Export-ModuleMember -Function Detach-HNSHostEndpoint
+Export-ModuleMember -Function Detach-HNSVMEndpoint
+Export-ModuleMember -Function Detach-HNSEndpoint
+
+Export-ModuleMember -Function Get-HNSPolicyList
+Export-ModuleMember -Function Remove-HnsPolicyList
+Export-ModuleMember -Function New-HnsRoute
+Export-ModuleMember -Function New-HnsLoadBalancer
+
+Export-ModuleMember -Function Invoke-HNSRequest
```
Comment (aravindhp) [inline tools/ansible/tasks/wsu/powershell/hns.psm1]: Please add newline

File: tools/ansible/tasks/wsu/main.yaml [hunk 4]
```diff
@@ -199,3 +204,26 @@
...
+      win_copy:
+        src: "powershell/hns.psm1"
+        dest: "{{ win_temp_dir.path }}\\hns.psm1"
+
+    - name: Get source-vip
+      win_shell: 'Import-Module {{ win_temp_dir.path }}\\hns.psm1; $net = (Get-HnsNetwork | where { $_.Name -eq "OpenShiftNetwork" }); $endpoint = New-HnsEndpoint -NetworkId $net.ID -Name VIPEndpoint; Attach-HNSHostEndpoint -EndpointID $endpoint.ID -CompartmentID 1; (Get-NetIPConfiguration -AllCompartments -All -Detailed | where { $_.NetAdapter.LinkLayerAddress -eq $endpoint.MacAddress }).IPV4Address.IPAddress.Trim()'
+      register: source_vip
+
+    - name: Create kube-proxy Windows Service
+      win_service:
```
Comment (aravindhp) [inline tools/ansible/tasks/wsu/main.yaml]: If the node is rebooted, is the service configured to start? If not, please ensure that it does.

File: tools/ansible/tasks/wsu/powershell/hns.psm1 (hunk 2 above)
Comment (sebsoto) [reply tools/ansible/tasks/wsu/powershell/hns.psm1]: I copied the whole file, just changed it to only the functions we needed.

Comment (aravindhp) [reply tools/ansible/tasks/wsu/powershell/hns.psm1]: I am confused. Did you copy the whole file without any changes or did you just copy the function that we need? Either way it needs to be attributed better and we need to check if the license matches up.

Comment (sebsoto) [reply tools/ansible/tasks/wsu/powershell/hns.psm1]: I copied the whole file originally and added the MIT license, which is what the file was under, to the top of the file.
I'm under the impression that the Apache 2.0 license is compatible with the MIT license, but thats just what google told me, not a lawyer. In the revisions commit I then removed the parts of the file that we did not need.

**Comment by sebsoto** [comment]
@aravindhp squashed

**Pull request merged**